c++/src/Compression.cc

```diff
@@ -636,6 +884,33 @@ DIAGNOSTIC_POP
     return static_cast<uint64_t>(result);
   }
 
+  std::unique_ptr<BufferedOutputStream>
+     createCompressor(
+                      CompressionKind kind,
+                      OutputStream * outStream,
+                      CompressionStrategy strategy,
+                      uint64_t bufferCapacity,
+                      uint64_t blockSize,
+                      MemoryPool& pool) {
+    switch (static_cast<int64_t>(kind)) {
+    case CompressionKind_NONE: {
+      return std::unique_ptr<BufferedOutputStream>
+        (new BufferedOutputStream(pool, outStream, bufferCapacity, blockSize));
+    }
+    case CompressionKind_ZLIB: {
+      int level = (strategy == CompressionStrategy_SPEED) ? -1 : 9;
```

Comment:
Compare the three options, `CompressionStrategy_DEFAULT`, `CompressionStrategy_SPEED`, `CompressionStrategy_COMPRESSION` and set the level accordingly.